app/models/pet.rb

```diff
@@ -2,12 +2,18 @@ class Pet < ApplicationRecord
   validates :name, presence: true
   validates :age, presence: true, numericality: true
   belongs_to :shelter
-
+  has_many :application_pets
+  has_many :applications, through: :application_pets
+  
   def shelter_name
     shelter.name
   end
 
   def self.adoptable
     where(adoptable: true)
   end
+
+  def find_application_pet(application_id)
+    ApplicationPet.where("application_pets.application_id = #{application_id} AND application_pets.pet_id = #{self.id}").first
+  end
```

Comment:
Based on the name of this method it looks like you are intending to retrieve a pet's application. In your controller I made a suggestion as to how you can do this much easier using `find_by`.